devtools/server/actors/replay/module.js

```diff
@@ -26,7 +26,11 @@ const { Debugger, RecordReplayControl, Services, InspectorUtils } = sandbox;
 const isRecordingOrReplaying = !!RecordReplayControl.onNewSource;
 
 if (isRecordingOrReplaying) {
-  Services.cpmm.sendAsyncMessage("RecordingStarting");
+  const recordingId = RecordReplayControl.recordingId();
```

Comment:
The main downside of this is that this will block page load until we get a response from our servers. In an ideal world, I think we'd want to be able to give the driver a callback, and have the driver call the callback once it knows the recording ID.

Reply:
ahh - that doesn't seem ideal

Could we also put this in a timeout?

```js
setImmediate(() => {
  const recordingId = RecordReplayControl.recordingId();

})
```

If not - yeah, lets just add the driver callback

Reply:
That's fair. `setImmediate` is probably too soon to make a difference. Maybe we could have a `setTimeout` that waits 5 seconds and then calls `RecordReplayControl.recordingId();` so we just assume that is enough time to not block enough to be a problem. 

> If not - yeah, lets just add the driver callback

I have no sense for how complicated that would be, so that's probably the main question. Sounds like we'd need a C++ callback that queued up a job to call the callback.

@bhackett1024 Do you have any thoughts on this discussion and approaches?

Reply:
We actually have a RecordReplaySetRecordingIdCallback driver API which does something similar and which gecko-dev already uses, but it's kind of tricky to use because it's called off the main thread and only when recording.

I don't understand the context here, though.  Why is it reasonable for RecordingFinished to take 5 seconds to deliver?  It seems like this should happen more or less immediately, because the RecordingFinished notification is sent before the main thread blocks and waits for the recording to finish uploading.

Reply:
> We actually have a RecordReplaySetRecordingIdCallback driver API which does something similar and which gecko-dev already uses, but it's kind of tricky to use because it's called off the main thread and only when recording.

Oh hah well that makes life potentially easier at least.

> I don't understand the context here, though. Why is it reasonable for RecordingFinished to take 5 seconds to deliver? It seems like this should happen more or less immediately, because the RecordingFinished notification is sent before the main thread blocks and waits for the recording to finish uploading.

That's a good question, let me revisit.